File: backend_samples/library/dotnet/AzureAIVisionFaceDeviceAttestation/src/Android/AndroidVerifier.cs
using Azure.AI.Vision.Face.DeviceAttestation.Services;

namespace Azure.AI.Vision.Face.DeviceAttestation.Android;

/// <summary>
/// Android Play Integrity + Key Attestation verifier — the seam the register
/// handler depends on. Delegates to <see cref="AndroidVerification"/>.
/// </summary>
internal static class AndroidVerifier
{
    /// <summary>Verify Android Key Attestation + Play Integrity.</summary>
    public static Task<AuthVerificationResult> VerifyAsync(AttestationContext ctx, AttestationMessageData messageData, string attestJson)
        => AndroidVerification.VerifyAsync(ctx.Config, ctx.Logger, messageData, attestJson);
}

File: backend_samples/library/java/AzureAIVisionFaceDeviceAttestation/src/main/java/com/azure/ai/vision/face/deviceattestation/android/AndroidVerifier.java
package com.azure.ai.vision.face.deviceattestation.android;

import com.azure.ai.vision.face.deviceattestation.AttestationContext;
import com.azure.ai.vision.face.deviceattestation.services.AttestationMessageData;
import com.azure.ai.vision.face.deviceattestation.services.AuthVerificationResult;

/**
 * Android Play Integrity + Key Attestation verifier — the seam the register
 * handler depends on. Delegates to {@link AndroidVerification}.
 */
public final class AndroidVerifier {

    /** Verify Android Key Attestation + Play Integrity. */
    public static AuthVerificationResult verify(AttestationContext ctx, AttestationMessageData messageData, String attestJson) {
        return AndroidVerification.verify(ctx.config, ctx.logger, messageData, attestJson);
    }

    private AndroidVerifier() {
    }
}
